Sources/Networking/Operations/Handling/CustomerInfoResponseHandler.swift

```diff
@@ -32,7 +65,27 @@ class CustomerInfoResponseHandler {
             }
             .mapError(BackendError.networkError)
 
-        completion(result)
+        self.handle(result: result, completion: completion)
+    }
+
+    private func handle(
+        result: Result<CustomerInfo, BackendError>,
+        completion: @escaping CustomerAPI.CustomerInfoResponseHandler
+    ) {
+        guard result.error?.isServerDown == true,
+        #available(iOS 15.0, tvOS 15.0, watchOS 8.0, macOS 12.0, *) else {
+            completion(result)
+            return
+        }
+
+        _ = Task<Void, Never> {
+            do {
+                completion(.success(try await self.computeOfflineCustomerInfo()))
+            } catch {
+                Logger.error(Strings.offlineEntitlements.computing_offline_customer_info_failed(error))
+                completion(result)
+            }
+        }
```

Comment:
The next time the app loads after having had entitlements computed offline, we'll be returning from cache first, right? Given that this only is used after a network request is made? 

If that's the case, then it's a bit of a problem - you make a purchase, get entitlements, next time you open the app the entitlements are gone until the notification comes in the delegate, and even then you'll only get them if the app listens to the delegate. 

We need to make sure that if offline entitlements were ever returned, we're still returning them until we get a response from the backend.